File: lib/features/employee/new_recruite/domain/recruitment_announcement.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:police_com/core/enums/announcement_status.dart';

part 'recruitment_announcement.freezed.dart';
part 'recruitment_announcement.g.dart';

@freezed
abstract class RecruitmentAnnouncement with _$RecruitmentAnnouncement {
  const factory RecruitmentAnnouncement({
    required String? announcementId,
    required String? announcementTitle,
    required String? announcementDescription,
    required int requiredApplicants,
    required String? round,
    required DateTime? createdDate,
    String? createdBy,
    required DateTime? expiryDate,
    @Default(AnnouncementStatus.created) AnnouncementStatus status,
    DateTime? postedDate,
    String? postedBy,
  }) = _RecruitmentAnnouncement;

  factory RecruitmentAnnouncement.fromJson(Map<String, dynamic> json) =>
      _$RecruitmentAnnouncementFromJson(json);
}

File: Backend_dot_net_c_sharp_models_and_eums_to_convert_to_dart /Models/Employee/NewRecruit/RecruitmentAnnouncement.cs
using PeerageHRIS.Enumeration;
using System.ComponentModel.DataAnnotations;

namespace PeerageHRIS.Models.Employee.NewRecruit
{
    public class RecruitmentAnnouncement
    {
        [Key]
        [Required]
        [Display(Name ="Announcement Id")]
        public string? AnnouncementId { get; set; }

        [Required]
        [Display(Name = "Announcement Title")]
        public string? AnnouncementTitle { get; set; }

        [Required]
        [Display(Name = "Description")]
        public string? AnnouncementDescription { get; set; }

        [Required]
        [Display(Name = "Required Applicants")]
        public int RequiredApplicants { get; set; }

        [Required]
        [Display(Name = "Round")]
        public string? Round { get; set; }

        [Display(Name = "Created Date")]
        [Required]
        [DataType(DataType.Date)]
        public DateTime? CreatedDate { get; set; }

        [Display(Name = "Created By")]
        public string? CreatedBy { get; set; }

        [Display(Name = "Expiry Date")]
        [Required]
        [DataType(DataType.Date)]
        public DateTime? ExpiryDate { get; set; }

        [Display(Name = "Status")]
        public AnnouncementStatus Status { get; set; } = AnnouncementStatus.Created;

        [Display(Name = "Posted Date")]
        [DataType(DataType.Date)]
        public DateTime? PostedDate { get; set; }

        [Display(Name = "Posted By")]
        public string? PostedBy { get; set; }
    }
}
